### DISCUSSION TITLE
SecretsManager `secretStringTemplate` not using variable

### DISCUSSION THREAD

**Discussion by nabeelvalley** [aws/aws-cdk#15383] [Q&A]
Hi, I'm trying to generate some credentials using SecretsManager but it seems to not be behaving as expected:

I am using the following code to generate a secret:

```ts
new sm.Secret(this, "DBAdminCredentials", {
      generateSecretString: {
        secretStringTemplate: JSON.stringify({
          username: this.dbUsername,
        }),
        generateStringKey: "password",
      },
    });
```

Where `this.dbUsername` is defined at a class level:

```ts
public readonly dbUsername = "dbusername";
```

According to my understanding the username should be present in the secret, but the resulting object is this (Via SecretsManager in the Console):

```json
{
    "password": "xxxxxxxxxx"
}
```

However, if I don't use the variable and put the username straight into the template it works fine:

```ts
new sm.Secret(this, "DBAdminCredentials", {
      generateSecretString: {
        secretStringTemplate: JSON.stringify({
          username: "dbusername",
        }),
        generateStringKey: "password",
      },
    });
```

I've also noticed a similar issue with the `id` passed to `sm.Secret`, but I have not investigated as completely

Not sure if this is a bug or expected behaviour - does this also apply to other constructs too - it's not what I would expect to happen in this context

**Comment by peterwoodworth** [comment]
Is it possible that `this.dbUsername` refers to a token when this behavior occurs? Or are you actually strictly hardcoding it as described
